desktop-widgets/mainwindow.cpp

```diff
@@ -88,12 +88,25 @@ MainWindow *MainWindow::m_Instance = NULL;
 
 extern "C" void showErrorFromC()
 {
+	// Show errors only if we are running in the GUI thread.
+	// If we're not in the GUI thread, let errors accumulate.
+	if (QThread::currentThread() != QCoreApplication::instance()->thread()) {
+		return;
+	}
```

Comment:
can you please remove all braces `{ }` if the `if` statement wraps a single line.
becomes:
```
if (QThread::currentThread() != QCoreApplication::instance()->thread())
    return;
```

applies to the rest of the patch too.

Reply:
NOTE: i know that we have this in a lot of places, but the idea is to keep future `if` additions consistent.